stacktrace.go

```diff
@@ -350,14 +353,51 @@ func callerFunctionName() string {
 	return baseName(callerFrame.Function)
 }
 
+func goMajorMinorVersion() (int, int, error) {
+	version := runtime.Version()
+
+	goVersionMajorMinorRegexp := regexp.MustCompile(`go(\d+)\.(\d+)`)
+	match := goVersionMajorMinorRegexp.FindStringSubmatch(version)
+
+	couldNotParseError := fmt.Errorf("could not parse the go runtime version from '%s'", version)
+
+	if len(match) == 0 {
+		return 0, 0, couldNotParseError
+	}
+
+	var major, minor int
+	var err error
+	if major, err = strconv.Atoi(match[1]); err != nil {
+		return 0, 0, couldNotParseError
+	}
+	if minor, err = strconv.Atoi(match[2]); err != nil {
+		return 0, 0, couldNotParseError
+	}
+
+	return major, minor, nil
+}
+
 // packageName returns the package part of the symbol name, or the empty string
 // if there is none.
 // It replicates https://golang.org/pkg/debug/gosym/#Sym.PackageName, avoiding a
```

Comment:
This change brings this function in line with the `pkg/debug/gosym` package that this was imported from: https://cs.opensource.google/go/go/+/refs/tags/go1.21.0:src/debug/gosym/symtab.go;l=64